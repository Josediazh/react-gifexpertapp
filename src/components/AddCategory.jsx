import PropTypes from 'prop-types';
import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');
  const onInpuValueChange = ({target}) => {
      setInputValue(target.value);
  };
  const onSubmit = (event) =>{
    event.preventDefault();
    if (inputValue.length <= 1) return ;
    onNewCategory(inputValue);
    setInputValue('');
  };

  return (
    <form aria-label='formAddCategory' onSubmit={onSubmit}>
      <input 
      type="text" 
      placeholder="Buscar gif" 
      onChange={onInpuValueChange}
      value={inputValue}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
