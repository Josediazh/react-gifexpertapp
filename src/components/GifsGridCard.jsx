import PropTypes from 'prop-types';

export const GifsGridCard = ({id,title,img}) => {
   
  return (
    <div className="card">
      <img src={img} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifsGridCard.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired
}


