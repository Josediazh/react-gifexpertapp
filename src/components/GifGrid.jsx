import { GifsGridCard } from "./GifsGridCard";
import { UseFetchGifs } from "../hooks/UseFetchGifs";
import PropTypes from 'prop-types';

export default function GifGrid({category}) { 

  const {gifs,isloading } = UseFetchGifs(category)

  return (
    <>
    <h3>{category}</h3>
      {isloading && (<h4>Cargando...</h4>)}
      <div className="card-grid">
        {
            gifs.map( ({id,title,imgurl}) => {
                return (
                  < GifsGridCard key={id} id={id} title={title} img={imgurl} />
                )
            })
        }
      </div>
    </>
  )
}


GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}