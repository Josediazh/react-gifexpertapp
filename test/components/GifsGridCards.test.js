import { render, screen } from "@testing-library/react";
import { GifsGridCard } from "../../src/components/GifsGridCard";

describe('Testting GifsGridCards ', () => {

    const title = 'Dokuro chan';
    const img = 'http://dokurochan.com/dokurochan.jpg';

    test('GifsGridCards snapshot', () => {

        const { container } = render(<GifsGridCard title={title} img={img} />);
        expect(container).toMatchSnapshot();

     })


     test('IMG render with alt attr', () => { 
        
        render(<GifsGridCard title={title} img={img} />);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(img);
        expect(alt).toBe(title);
        
      })
});