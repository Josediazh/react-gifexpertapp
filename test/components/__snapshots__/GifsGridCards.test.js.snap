// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testting GifsGridCards  GifsGridCards snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dokuro chan"
      src="http://dokurochan.com/dokurochan.jpg"
    />
    <p>
      Dokuro chan
    </p>
  </div>
</div>
`;
